fe/fe-core/src/main/java/org/apache/doris/analysis/BinaryPredicate.java

```diff
@@ -411,6 +420,19 @@ private Type getCmpType() throws AnalysisException {
             }
         }
 
+        if (canCompareIP(getChild(0).getType().getPrimitiveType(), getChild(1).getType().getPrimitiveType())) {
+            if ((getChild(0).getType().isIP() && getChild(1) instanceof StringLiteral)
+                    || (getChild(1).getType().isIP() && getChild(0) instanceof StringLiteral)
+                    || (getChild(0).getType().isIP() && getChild(1).getType().isIP())) {
+                if (getChild(0).getType().isIPv4() || getChild(1).getType().isIPv4()) {
```

Comment:
if IPV4 cmp IPV6, the getCmpType will return IPV4, is this the expected behavior?